example-pipelines/environments/Jenkinsfile

```
@@ -0,0 +1,117 @@
+  }
+  stages {
+    stage('TF init & validate') {
+      when { anyOf {branch "prod";branch "dev";changeRequest() } }
+      steps {
+        container('terraform') {
+          sh '''
+          if [[ $CHANGE_TARGET ]]; then
+            TARGET_ENV=$CHANGE_TARGET
+          else
+            TARGET_ENV=$BRANCH_NAME
+          fi
+
+          if [ -d "example-pipelines/environments/${TARGET_ENV}/" ]; then
+            cd example-pipelines/environments/${TARGET_ENV}
+            terraform init
+            terraform validate
+          else
+            for dir in example-pipelines/environments/*/
+            do 
+              cd ${dir}
+              env=${dir%*/}
+              env=${env#*/}
+              echo ""
+              echo "*************** TERRAFOM INIT and VALIDATE ******************"
+              echo "******* At environment: ${env} ********"
+              echo "*************************************************"
+              terraform init || exit 1
+              terraform validate || exit 1
+              cd ../../../
+            done
+          fi'''
+        }
+      }
+    }
+    stage('TF plan') {
+      when { anyOf {branch "prod";branch "dev";changeRequest() } }
+      steps {
+        container('terraform') {
+          sh '''
+          if [[ $CHANGE_TARGET ]]; then
+            TARGET_ENV=$CHANGE_TARGET
+          else
+            TARGET_ENV=$BRANCH_NAME
+          fi
+         
+          if [ -d "example-pipelines/environments/${TARGET_ENV}/" ]; then
+            cd example-pipelines/environments/${TARGET_ENV}
+            terraform plan
+          else
+            for dir in example-pipelines/environments/*/
+            do 
+              cd ${dir}
+              env=${dir%*/}
+              env=${env#*/}
+              echo ""
+              echo "*************** TERRAFOM PLAN ******************"
+              echo "******* At environment: ${env} ********"
+              echo "*************************************************"
+              terraform plan || exit 1
+              cd ../../../
+            done
+          fi'''
+        }
+      }
+    }
+    stage('Wait for Appproval') {
+      when { anyOf {branch "prod";branch "dev" } }
+      steps {
+        script {
+          def userInput = input(id: 'confirm', message: 'Apply Terraform?', parameters: [ [$class: 'BooleanParameterDefinition', defaultValue: false, description: 'Apply terraform', name: 'confirm'] ])
+        }
+      }
+    }
+    stage('TF Apply') {
+      when { anyOf {branch "prod";branch "dev" } }
+      steps {
+        container('terraform') {
+          sh '''
+          if [[ $CHANGE_TARGET ]]; then
+            TARGET_ENV=$CHANGE_TARGET
+          else
+            TARGET_ENV=$BRANCH_NAME
+          fi
+
+          if [ -d "example-pipelines/environments/${TARGET_ENV}/" ]; then
+            cd example-pipelines/environments/${TARGET_ENV}
+            terraform apply -input=false --auto-approve test.out
```

Since we are not doing plan --out
```suggestion
            terraform apply -input=false -auto-approve
```

---

updated